%YAML:1.0

#--------------------------------------------------------------------------------------------
# Camera Parameters. Adjust them!
#--------------------------------------------------------------------------------------------
Camera.type: "PinHole" # Currently only Pinhole Model is supported

# Camera calibration and distortion parameters (OpenCV) 
Camera.fx: 432.0433654785156
Camera.fy: 432.0433654785156
Camera.cx: 423.03424072265625
Camera.cy: 235.6173095703125

Camera.k1: -0.00859388429671526
Camera.k2: 0.04788520932197571
Camera.p1: -0.04412417113780975
Camera.p2: 0.008043156005442142

# Camera resolution
Camera.width: 848
Camera.height: 480

# Camera frames per second 
Camera.fps: 30.0

# Color order of the images (0: BGR, 1: RGB. It is ignored if images are grayscale)
Camera.RGB: 1

# Set this to 1 while using Fisheye Cameras like the T265. For now this distortion is not 
# supported so we undistort the raw image and use it for all computations
is_fisheye: 0

# Transformation from camera to body-frame (imu)
Tbc: !!opencv-matrix
   rows: 4
   cols: 4
   dt: f
   data: [1, 0, 0, 0.03,
          0, 1, 0, -0.007,
          0, 0, 1, -0.016,
          0, 0, 0, 1]


#--------------------------------------------------------------------------------------------
# Frontend COVINS Parameters
#--------------------------------------------------------------------------------------------

# 1 if Odometry is published in IMU frame. Set to 0 if published in Camera Frame
odom_in_imu_frame: 0

# Feature type. Currently only ORB and SIFT supported
extractor.type: 'ORB' 
extractor.nFeatures: 1000 # 1000 for ORB and 300 for SIFT

#--------------------------------------------------------------------------------------------
# KF genration Parameters
#--------------------------------------------------------------------------------------------

#Minimum translational distance in metres for considering a new KF
t_min: 0.15
#Minimum rotational angle in radians for considering a KF. A new KF is generated if either of the
# translational or rotational condition is satisfied
r_min: 0.15

#--------------------------------------------------------------------------------------------
# ORB Parameters 
#--------------------------------------------------------------------------------------------
# For Place Recogntion, ORB Features are used
# The following parameters are used for ORB features for Place recognition 
# and also for ORB features for Matching (if extractor.type is ORB)

# ORB Extractor: Number of features per image
ORBextractor.nFeaturesPR: 1000 # 1000

# ORB Extractor: Scale factor between levels in the scale pyramid
ORBextractor.scaleFactor: 1.2
# ORB Extractor: Number of levels in the scale pyramid
ORBextractor.nLevels: 8

# ORB Extractor: Fast threshold
# Image is divided in a grid. At each cell FAST are extracted imposing a minimum response.
# Firstly we impose iniThFAST. If no corners are detected we impose a lower value minThFAST
# You can lower these values if your images have low contrast
ORBextractor.iniThFAST: 20
ORBextractor.minThFAST: 7
